src/transformers/image_utils.py

```diff
@@ -162,9 +162,9 @@ def infer_channel_dimension_format(image: np.ndarray) -> ChannelDimension:
     else:
         raise ValueError(f"Unsupported number of image dimensions: {image.ndim}")
 
-    if image.shape[first_dim] in (1, 3):
+    if image.shape[first_dim] in (1, 3, 4):
         return ChannelDimension.FIRST
-    elif image.shape[last_dim] in (1, 3):
+    elif image.shape[last_dim] in (1, 3, 4):
```

Comment:
@amyeroberts ViTMatte uses 4 input channels rather than the common 1 or 3. This made me make the updates here. However this causes some tests of tests/utils/test_image_utils.py to fail:

```
FAILED tests/utils/test_image_utils.py::UtilFunctionTester::test_get_channel_dimension_axis - AssertionError: 0 != 2
FAILED tests/utils/test_image_utils.py::UtilFunctionTester::test_infer_channel_dimension - AssertionError: <ChannelDimension.FIRST: 'channels_first'> != <ChannelDimension.LAST: 'channels_last'>
```
Can we keep this and update the tests accordingly, or should I revert the changes here?

Reply:
Let's keep this as is for the moment whilst this PR is still open. I'm currently working on making this more flexible for the image processors, which will hopefully be merged in soon.

Reply:
Looks like I can use #25462 for this purpose